slasher/cache/span_cache.go

```diff
@@ -0,0 +1,73 @@
+package cache
+
+import (
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prysmaticlabs/prysm/slasher/detection/attestations/types"
+)
+
+var (
+	// epochSpansCacheSize defines the max number of epoch spans the cache can hold.
+	epochSpansCacheSize = 256
+	// Metrics
+	epochSpansCacheHit = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "epoch_spans_cache_hit",
+		Help: "The total number of cache hits on the epoch spans cache.",
+	})
+	epochSpansCacheMiss = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "epoch_spans_cache_miss",
+		Help: "The total number of cache misses on the epoch spans cache.",
+	})
+)
+
+// EpochSpansCache is used to store the spans needed on a per-epoch basis for slashing detection.
+type EpochSpansCache struct {
+	cache *lru.Cache
+}
+
+// NewEpochSpansCache initializes the map and underlying cache.
+func NewEpochSpansCache() *EpochSpansCache {
+	cache, err := lru.New(epochSpansCacheSize)
+	if err != nil {
+		panic(err)
+	}
+	return &EpochSpansCache{cache: cache}
+}
+
+// Get returns a cached response given the requested epoch, if any.
+func (c *EpochSpansCache) Get(epoch uint64) (map[uint64]types.Span, bool) {
+	item, exists := c.cache.Get(epoch)
+	if exists && item != nil {
+		epochSpansCacheHit.Inc()
+		return item.(map[uint64]types.Span), true
+	}
+
+	epochSpansCacheMiss.Inc()
+	return make(map[uint64]types.Span), false
+}
+
+// Set the response in the cache.
+func (c *EpochSpansCache) Set(epoch uint64, epochSpans map[uint64]types.Span) {
+	_ = c.cache.Add(epoch, epochSpans)
+}
+
+// Delete removes an epoch from the cache and returns if it existed or not.
+func (c *EpochSpansCache) Delete(epoch uint64) (present bool) {
```

Comment:
why `(present bool)`? instead of bool?, doesn't look like the standard convention to me

Reply:
Might be confusing what the return bools mean without context, but it's mentioned in the comment too